src/Magento/FunctionalTestingFramework/Test/Util/ActionMergeUtil.php

```diff
@@ -106,13 +106,16 @@ private function resolveSecretFieldAccess($resolvedActions)
         $actions = [];
         foreach ($resolvedActions as $resolvedAction) {
             $action = $resolvedAction;
-            $hasSecretRef = $this->actionAttributeContainsSecretRef($resolvedAction->getCustomActionAttributes());
+            $actionHasSecretRef = $this->actionAttributeContainsSecretRef($resolvedAction->getCustomActionAttributes());
+            $dataHasSecretRef = $this->dataFieldContainsSecretRef($resolvedAction->getCustomActionAttributes());
+            $approvedActions = array('fillField', 'magentoCLI', 'field');
+            $actionType = $resolvedAction->getType();
 
-            if ($resolvedAction->getType() !== 'fillField' && $hasSecretRef) {
-                throw new TestReferenceException("You cannot reference secret data outside of fill field actions");
+            if (!(in_array($resolvedAction->getType(), $approvedActions)) && ($actionHasSecretRef || $dataHasSecretRef)) {
```

Comment:
Change conditional to just use `actionType` instead of accessing getter function.

Reply:
Done.